### DISCUSSION TITLE
regex only checks if second char is written in input

### DISCUSSION THREAD

**Discussion by philmetscher** [neuefische/questions#282] [Web] [answered]
Heyho,
I am currently working on a fix where I have to check first char of an input field is not a space.
I am currently trying to solve this with a regex that is supposed to check this. However, this only goes into the if-statement when a second char is entered.
Regex is a complicated thing. Maybe someone can help me here :)

```javascript
  const startsWith = new RegExp("^[^-\\s][\\w]");

  //will be triggered by an onChange
  function handleCategoryInput(event) {
    if (startsWith.test(event.target.value)) {
        console.log("if");
        ...
    } else {
      console.log("else")
    }
  }
```

If you need more Infos let me know 👍

**Comment by pixelass** [comment, marked as answer]
If you need to check if a string starts with a certain character, you can use [`String.prototype.startsWith()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)

```js
const input1 = " foo";
const input2 = "foo";

console.log(input1.startsWith(" ")); // true
console.log(input2.startsWith(" ")); // false
```

Example: https://codepen.io/dekk/pen/bGKGxKr/9fcfafe916bab6c28e0a6e9f8f3b01f7?editors=0011

**Reply by pixelass** [reply]
If you want to use regex (there is really no need here), you can do this:

```js
console.log(Boolean(/^\s/.exec(" foo"))); // true
console.log(Boolean(/^\s/.exec("foo"))); // false


function startsWith(value, testString) {
  return Boolean(new RegExp(`^${testString}`).exec(value))
}

console.log(startsWith(" foo", " ")); // true
console.log(startsWith("foo", " ")); // false

console.log(startsWith("abcdefg", "abc")); // true

// This is the recommended version:
console.log("abcdefg".startsWith("abc")); // true
```

Example: https://codepen.io/dekk/pen/VwdwGqv/019a760470d0ee8723d67d7490bd67b4?editors=1011

**Reply by philmetscher** [reply]
THANKS!

**Reply by philmetscher** [reply]
I think it was a blocker in my head 😅 
I tried it before with startsWith but i doesn't work at another place
